test/system/src/test/java/io/pravega/test/system/BookieFailoverTest.java

```diff
@@ -144,40 +145,50 @@ public void tearDown() {
     }
 
     @Test
-    public void bookieFailoverTest() throws ExecutionException, InterruptedException {
+    public void bookieFailoverTest() throws Exception {
         createWriters(clientFactory, NUM_WRITERS, SCOPE, STREAM);
         createReaders(clientFactory, readerGroupName, SCOPE, readerGroupManager, STREAM, NUM_READERS);
 
         // Give some time to create readers before forcing a bookie failover.
-        Exceptions.handleInterrupted(() -> Thread.sleep(BOOKIE_FAILOVER_WAIT_MILLIS));
+        AssertExtensions.assertEventuallyEquals("Writers and/or readers not progressing.", true,
+            () -> testState.getEventWrittenCount() > 100 && testState.getEventReadCount() > 100, 1000, IO_PROGRESS_CHECK_TIMEOUT);
```

Comment:
Parameter value = 100, better if have it configurable. I think so, but may not be required.

Reply:
Done.